tests/integration/release/test_release.py

```diff
@@ -35,17 +41,117 @@ def test_column_noising(
+
+    if expected_noise != "default":
+        config_dict[dataset_name][Keys.ROW_NOISE][NOISE_TYPES.omit_row.name][
+            Keys.ROW_PROBABILITY
+        ] = expected_noise
+        config = NoiseConfiguration(LayeredConfigTree(config_dict))
+    else:
+        config = NoiseConfiguration(LayeredConfigTree(config_dict))
+        expected_noise: float = config.get_row_probability(
+            dataset_name, NOISE_TYPES.omit_row.name
+        )
+    dataset_schema = DATASET_SCHEMAS.get_dataset_schema(dataset_name)
+    dataset = Dataset(dataset_schema, original_data, 0)
+    NOISE_TYPES.omit_row(dataset, config)
+    noised_data = dataset.data
+
+    fuzzy_checker.fuzzy_assert_proportion(
+        name="test_omit_row",
+        observed_numerator=len(noised_data),
+        observed_denominator=len(original_data),
+        target_proportion=1 - expected_noise,
+    )
+
+    assert set(noised_data.columns) == set(original_data.columns)
+    assert (noised_data.dtypes == original_data.dtypes).all()
 
-    run_omit_row_or_do_not_respond_tests(dataset_name, config, original_data, noised_data)
+
+@pytest.mark.parametrize("expected_noise", ["default", 0.03])
+def test_do_not_respond(
+    expected_noise: str | float,
+    unnoised_dataset: Dataset,
+    dataset_name: str,
+    fuzzy_checker: FuzzyChecker,
+) -> None:
+    """Tests that do_not_respond noising is being applied at the expected proportion."""
+    if dataset_name not in [
+        DATASET_SCHEMAS.acs.name,
+        DATASET_SCHEMAS.census.name,
+        DATASET_SCHEMAS.cps.name,
+    ]:
+        return
+
+    original_data = unnoised_dataset.data
+    config_dict = get_single_noise_type_config(dataset_name, NOISE_TYPES.do_not_respond.name)
+
+    if expected_noise != "default":
+        config_dict[dataset_name][Keys.ROW_NOISE][NOISE_TYPES.do_not_respond.name][
+            Keys.ROW_PROBABILITY
+        ] = expected_noise
+        config = NoiseConfiguration(LayeredConfigTree(config_dict))
+    else:
+        config = NoiseConfiguration(LayeredConfigTree(config_dict))
+        expected_noise: float = config.get_row_probability(
+            dataset_name, NOISE_TYPES.do_not_respond.name
+        )
+
+    dataset_schema = DATASET_SCHEMAS.get_dataset_schema(dataset_name)
+    dataset = Dataset(dataset_schema, original_data, 0)
+    NOISE_TYPES.do_not_respond(dataset, config)
+    noised_data = dataset.data
+
+    # Check ACS and CPS data is scaled properly due to oversampling
+    if dataset_name is not DATASET_SCHEMAS.census.name:  # is acs or cps
+        expected_noise = 0.5 + expected_noise / 2
+
+    # Test that noising affects expected proportion with expected types
+    fuzzy_checker.fuzzy_assert_proportion(
+        name="test_do_not_respond",
+        observed_numerator=len(original_data) - len(noised_data),
+        observed_denominator=len(original_data),
+        target_proportion=expected_noise,
+        name_additional=f"noised_data",
```

Comment:
You have this here but not in the other test. Why?

Reply:
Looking at the docstring for fuzzy_assert_proportion, it looks like this option will print this name to diagnostic output. Not sure why it's here and not the other check, but I'll invoke Chesterton's fence and keep it as is.

Reply:
Note: I also updated the observed_numerator, observed_denominator, and target_proportion to be the same in both tests. They were previously defined differently but mathematically doing the same thing, as in 1-(N/O) = (O - N) / O